client/genericclient/client.go

```diff
@@ -31,12 +31,15 @@ var _ Client = &genericServiceClient{}
 
 // NewClient create a generic client
 func NewClient(destService string, g generic.Generic, opts ...client.Option) (Client, error) {
-	svcInfo := generic.ServiceInfoWithCodec(g)
+	svcInfo := generic.ServiceInfoWithGeneric(g)
 	return NewClientWithServiceInfo(destService, g, svcInfo, opts...)
 }
 
 // NewClientWithServiceInfo create a generic client with serviceInfo
 func NewClientWithServiceInfo(destService string, g generic.Generic, svcInfo *serviceinfo.ServiceInfo, opts ...client.Option) (Client, error) {
+	if isDeprecated, ok := svcInfo.Extra["deprecated"].(bool); ok && isDeprecated {
```

Comment:
It's better to define a unique constant key, like `deprecated_generic_service_info_api`